search-solr/src/main/java/edu/unc/lib/boxc/search/solr/services/SolrSearchService.java

```diff
@@ -562,14 +562,14 @@ private void addRangeFields(SearchState searchState, SolrQuery query) {
         }
     }
 
-    private String getRangeValue(String key, String value) {
+    private String getRangeValue(String key, String value, boolean endRange) {
         if (StringUtils.isBlank(value)) {
             return "*";
         }
 
         if (SearchSettings.FIELDS_DATE_SEARCHABLE.contains(key)) {
             try {
-                return DateFormatUtil.getFormattedDate(value, true, true);
+                return DateFormatUtil.getFormattedDate(value, true, endRange);
```

Comment:
what is the end range used for?

Reply:
the DateFormatUtil takes the input in "YYYY" format and converts it into a timestamp format, which solr requires. As a result, it needs to add a month, day, etc, to the year. Since we are doing an inclusive search here, any dates that occur in the start or end year need to be included. So if we're searching for "2020,2022", it needs to search solr for the range "2020-01-01T00:00:00Z to 2022-12-31T23:59.59". The "endRange" flag tells it whether to expand to the beginning of the year or the end of the year.

Reply:
hmm, then maybe you could rename it to `includeEndOfYear`?

Reply:
hmm, its slightly more complicated since it interacts with the inclusive flag as well. For example, if inclusive=false then "2020,2022" would become "2020-01-01T23:59:59.999Z to 2021-12-31T23:59:59.999Z". 

The tests help make it clearer what happens:
https://github.com/UNC-Libraries/box-c/blob/main/search-solr/src/test/java/edu/unc/lib/boxc/search/solr/utils/DateFormatUtilTest.java#L61

This method is super complicated though and could use a refactor, not to mention that this is the only place it is used and the `inclusive` flag is never false now in practice (it was used at one point).

Reply:
i suppose the tests would make it clear if i could remember what each boolean passed in was when i looked at them =P
ok then